### PR TITLE
wipe search history when we apply new connection settings

### PR DESCRIPTION
**Pull request by kdrtibor**

### CODE REVIEW

File: src/main/java/com/hpe/adm/octane/ideplugins/intellij/settings/ConnectionSettingsConfigurable.java [hunk 1]
```diff
@@ -146,6 +148,10 @@ public void apply() throws ConfigurationException {
         ConnectionSettings newConnectionSettings = null;
         try {
             newConnectionSettings = testConnection();
+            //We should not have search history from any previous workspaces
+            new Thread(()-> {
```
Comment (atothhpe) [inline src/main/java/com/hpe/adm/octane/ideplugins/intellij/settings/ConnectionSettingsConfigurable.java]: Why is this in a thread?

Comment (kdrtibor) [reply src/main/java/com/hpe/adm/octane/ideplugins/intellij/settings/ConnectionSettingsConfigurable.java]: Well what it does is clear the search history, its a task that can easily be run in the background and not intended to slow down the log.

File: src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/detail/EntityDetailPresenter.java [hunk 2]
```diff
@@ -111,7 +110,9 @@ public void setEntity(Entity entityType, Long entityId) {
                         entityModel.setValue(new StringFieldModel(DetailsViewDefaultFields.FIELD_DESCRIPTION, description));
 
                         return entityModel;
-                    } catch (ServiceException ex) {
+                    } catch (Exception ex) {
+                        //Exception made for branch built
```
Comment (atothhpe) [inline src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/detail/EntityDetailPresenter.java]: You will never change back :D it's staying like this

Comment (kdrtibor) [reply src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/detail/EntityDetailPresenter.java]: So is it okay like this and i should remove the todo?

File: src/main/java/com/hpe/adm/octane/ideplugins/intellij/settings/ConnectionSettingsConfigurable.java (hunk 1 above)
Comment (atothhpe) [reply src/main/java/com/hpe/adm/octane/ideplugins/intellij/settings/ConnectionSettingsConfigurable.java]: does clearing the history slow down anything? it should be a very fast operation, it edits a file. I think making a thread for it is too much overhead

File: src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/detail/EntityDetailPresenter.java (hunk 2 above)
Comment (atothhpe) [reply src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/detail/EntityDetailPresenter.java]: y

File: src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/searchresult/SearchHistoryManager.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                jsonObject);
+    }
+
+    public void addToSearchHistory(String string) {
+        if (searchHistory.contains(string)) {
+            searchHistory.remove(string);
+        }
+        searchHistory.add(0, string);
+        if (searchHistory.size() > HISTORY_SIZE) {
+            searchHistory.remove(HISTORY_SIZE);
```
Comment (paul-ciobanu) [inline src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/searchresult/SearchHistoryManager.java]: @kdrtibor This doesn't seem to be correct

File: src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/searchresult/SearchHistoryManager.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    private IdePluginPersistentState idePluginPersistentState;
+
+    private List<String> searchHistory = new ArrayList<>();
+
+    private final int HISTORY_SIZE = 5;
+
+    public SearchHistoryManager(){
+    }
+
+    public void saveSearchHistory() {
```
Comment (paul-ciobanu) [inline src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/searchresult/SearchHistoryManager.java]: @kdrtibor Why do you need explicit/public save and load history methods?

**Pull request merged**